In [1]:
from qiskit_ibm_provider import IBMProvider
from qiskit.transpiler.coupling import CouplingMap

In [2]:
try:
    print(provider)
except NameError:
    provider = IBMProvider()

In [3]:
ibm_backend_name = 'ibm_lagos'
backend = provider.get_backend(ibm_backend_name)
config = backend.configuration()
props = backend.properties()

# dt to nanoseconds

In [4]:
config.dt

2.2222222222222221e-10

# Single qubit gate durations

## Code

In [5]:
config.basis_gates

['cx', 'id', 'rz', 'sx', 'x']

In [6]:
for gate in ['id', 'sx', 'x']:
    print("{} length in seconds: {}. ".format(gate, props.gate_length(gate, (1))))
    print("{} length in dt: {}. ".format(gate, props.gate_length(gate, (1)) / config.dt ))

id length in seconds: 3.5555555555555554e-08. 
id length in dt: 160.0. 
sx length in seconds: 3.5555555555555554e-08. 
sx length in dt: 160.0. 
x length in seconds: 3.5555555555555554e-08. 
x length in dt: 160.0. 


## Analysis

Each single qubit gate takes $35.555555555555554$ ns or $160$ dt.

# CX Duration

In [20]:
for q1, q2 in config.gates[4].coupling_map:
    cx_len = props.gate_length('cx', (q1, q2))
    cx_len_ns = cx_len * 1e+9
    print("- {} {:.3f} ns = {} dt".format((q1, q2), cx_len_ns, cx_len/ config.dt))

- (0, 1) 576.000 ns = 2592.0 dt
- (1, 0) 611.556 ns = 2752.0 dt
- (1, 2) 327.111 ns = 1472.0 dt
- (1, 3) 334.222 ns = 1504.0 dt
- (2, 1) 291.556 ns = 1312.0 dt
- (3, 1) 298.667 ns = 1344.0 dt
- (3, 5) 960.000 ns = 4320.0 dt
- (4, 5) 362.667 ns = 1632.0 dt
- (5, 3) 995.556 ns = 4480.000000000001 dt
- (5, 4) 327.111 ns = 1472.0 dt
- (5, 6) 256.000 ns = 1152.0000000000002 dt
- (6, 5) 291.556 ns = 1312.0 dt


# Readout duration

In [8]:
config.meas_levels

[1, 2]

In [12]:
for i in range(7):
    print(i, "duration in seconds: ", props.readout_length(i))
    print(i, "duration in dt: ", props.readout_length(i) / config.dt )

0 duration in seconds:  7.893333333333333e-07
0 duration in dt:  3552.0
1 duration in seconds:  7.893333333333333e-07
1 duration in dt:  3552.0
2 duration in seconds:  7.893333333333333e-07
2 duration in dt:  3552.0
3 duration in seconds:  7.893333333333333e-07
3 duration in dt:  3552.0
4 duration in seconds:  7.893333333333333e-07
4 duration in dt:  3552.0
5 duration in seconds:  7.893333333333333e-07
5 duration in dt:  3552.0
6 duration in seconds:  7.893333333333333e-07
6 duration in dt:  3552.0


Readout duration: $789.3333333333333$ ns

# Final Results

- Single qubit gate: $35.555555555555554$ ns = $160$ dt
- Readout duration: $789.3333333333333$ ns = $3552$ dt

## CNOT durations:

- (0, 1) 576.000 ns = 2592.0 dt
- (1, 0) 611.556 ns = 2752.0 dt
- (1, 2) 327.111 ns = 1472.0 dt
- (1, 3) 334.222 ns = 1504.0 dt
- (2, 1) 291.556 ns = 1312.0 dt
- (3, 1) 298.667 ns = 1344.0 dt
- (3, 5) 960.000 ns = 4320.0 dt
- (4, 5) 362.667 ns = 1632.0 dt
- (5, 3) 995.556 ns = 4480.000000000001 dt
- (5, 4) 327.111 ns = 1472.0 dt
- (5, 6) 256.000 ns = 1152.0000000000002 dt
- (6, 5) 291.556 ns = 1312.0 dt